hera_cal/reflections.py

```diff
@@ -51,169 +72,80 @@
                 within for reflections. Must be either both positive or both negative.
+            flags : DataContainer, flags of data. Default is None.
             keys : list of len-3 tuples, keys in data model reflections over. Default
                 is to use all auto-correlation keys.
-            data : str, data dictionary to find reflections in. Options are ['clean', 'data'].
-            edgecut : int, number of channels on band edges to exclude from modeling
-            taper : str, tapering function to apply across freq before FFT
+            edgecut_low : int, Nbins to flag but not window at low-side of the FFT axis.
+            edgecut_hi : int, Nbins to flag but not window at high-side of the FFT axis.
+            window : str, tapering function to apply across freq before FFT
             alpha : float, if taper is Tukey, this is its alpha parameter
-            zero_pad : int, number of channels to pad band edges with zeros before FFT
+            zeropad : int, number of channels to pad band edges with zeros before FFT
             fthin : int, scaling factor to down-select frequency axis when solving for phase
+            ref_sig_cut : float, if max reflection significance is not above this, do not record solution
             overwrite : bool, if True, overwrite dictionaries
+
+        Result:
+            self.ref_eps : dict, reflection epsilon term
+            self.ref_amp : dict, reflection amplitude
+            self.ref_phs : dict, reflection phase [radians]
+            self.ref_dly : dict, reflection delay [nanosec]
+            self.ref_flags : dict, reflection flags [bool]
+            self.ref_significance : dict, reflection significance,
+                which is max|Vfft| / median|Vfft| near reflection peak.
         """
         # initialize containers
         if hasattr(self, 'epsilon') and not overwrite:
             raise ValueError("reflection dictionaries exist but overwrite is False...")
-        self.epsilon = {}
-        self.amps = {}
-        self.phs = {}
-        self.delays = {}
-        self.peak_ratio = {}
-        self.dfft = DataContainer({})
-
-        # configure data
-        if data == 'clean':
-            data = self.clean_data
-            freqs = self.clean_freqs
-            self.reflection_times = self.clean_times
-        elif data == 'data':
-            # if using unCLEANed data, apply flags now on-the-fly
-            data = DataContainer(dict([(k, self.data[k] * ~self.flags[k]) for k in self.data.keys()]))
-            freqs = self.freqs
-            self.reflection_times = self.times
-        else:
-            raise ValueError("Didn't recognize data dictionary {}".format(data))
-
-        # get keys: only use auto correlations to model reflections
+        self.ref_eps = {}
+        self.ref_amp = {}
+        self.ref_phs = {}
+        self.ref_dly = {}
+        self.ref_flags = {}
+        self.ref_significance = {}
+
+        # get flags
+        if flags is None:
```

Comment:
Because the data is supposed to be CLEANed, and `self.flags` has all of the non-factorized flags in them. The `flags` kwarg should be the CLEAN flags. I'll update the argument names to be more clear about this!